libr/debug/p/native/windows/windows_debug.c

```diff
@@ -0,0 +1,1297 @@
+		perror ("__w32_info_user/malloc usr");
+		goto err___w32_info_user;
+	}
+	*usr = '\0';
+	usr_dom = (LPTSTR)malloc (usr_dom_len);
+	if (!usr_dom) {
+		perror ("__w32_info_user/malloc usr_dom");
+		goto err___w32_info_user;
+	}
+	*usr_dom = '\0';
+	if (!LookupAccountSid (NULL, tok_usr->User.Sid, usr, &usr_len, usr_dom, &usr_dom_len, &snu)) {
+		r_sys_perror ("__w32_info_user/LookupAccountSid");
+		goto err___w32_info_user;
+	}
+	if (*usr_dom) {
+		rdi->usr = r_str_newf (W32_TCHAR_FSTR"\\"W32_TCHAR_FSTR, usr_dom, usr);		
+	} else {
+		rdi->usr = r_sys_conv_win_to_utf8 (usr);
+	}
+err___w32_info_user:
+	if (h_tok) {
+		CloseHandle (h_tok);
+	}
+	free (usr);
+	free (usr_dom);
+	free (tok_usr);
+}
+
+static void __w32_info_exe(RDebug *dbg, RDebugInfo *rdi) {
+	RIOW32Dbg *rio = dbg->user;
+	rdi->exe = __resolve_path (rio->ph);
+#if 0
+	HANDLE ph = OpenProcess (PROCESS_QUERY_INFORMATION, FALSE, dbg->pid);
+	if (!ph) {
+		r_sys_perror ("__w32_info_exe/OpenProcess");
+		return;
+	}
+	rdi->exe = __resolve_path (ph);
+	CloseHandle (ph);
+#endif
+}
+
+RDebugInfo *w32_info(RDebug *dbg, const char *arg) {
+	RDebugInfo *rdi = R_NEW0 (RDebugInfo);
+	rdi->status = R_DBG_PROC_SLEEP; // TODO: Fix this
+	rdi->pid = dbg->pid;
+	rdi->tid = dbg->tid;
+	rdi->lib = (void *) __r_debug_get_lib_item();
+	rdi->thread = (void *)__r_debug_get_thread_item ();
+	rdi->uid = -1;
+	rdi->gid = -1;
+	rdi->cwd = NULL;
+	rdi->exe = NULL;
+	rdi->cmdline = NULL;
+	rdi->libname = NULL;
+	__w32_info_user (dbg, rdi);
+	__w32_info_exe (dbg, rdi);
+	return rdi;
+}
+
+static RDebugPid *__build_debug_pid(int pid, HANDLE ph, const char* name) {
+	char *path = NULL;
+	int uid = -1;
+	if (!ph) {
+		ph = OpenProcess (PROCESS_QUERY_INFORMATION | PROCESS_VM_READ, FALSE, pid);
+		if (ph) {
+			path = __resolve_path (ph);
+			DWORD sid;
+			if (ProcessIdToSessionId (pid, &sid)) {
+				uid = sid;
+			}
+			CloseHandle (ph);
+		} else {
+			return NULL;
+		}
+	} else {
+		path = __resolve_path (ph);
+		DWORD sid;
+		if (ProcessIdToSessionId (pid, &sid)) {
+			uid = sid;
+		}
+	}
+	if (!path) {
+		path = r_sys_conv_win_to_utf8 (name);
+	}
+	// it is possible to get pc for a non debugged process but the operation is expensive and might be risky
+	RDebugPid *ret = r_debug_pid_new (path, pid, uid, 's', 0);
+	free (path);
+	return ret;
+}
+
+RList *w32_pid_list(RDebug *dbg, int pid, RList *list) {
+	HANDLE sh = CreateToolhelp32Snapshot (TH32CS_SNAPPROCESS, pid);
+	if (sh == INVALID_HANDLE_VALUE) {
+		r_sys_perror ("w32_pid_list/CreateToolhelp32Snapshot");
+		return list;
+	}
+	PROCESSENTRY32 pe;
+	pe.dwSize = sizeof (pe);
+	if (Process32First (sh, &pe)) {
+		RIOW32Dbg *rio = dbg->user;
+		bool all = pid == 0, b = false;
+		do {
+			if (all || pe.th32ProcessID == pid || (b = pe.th32ParentProcessID == pid)) {
+				// Returns NULL if process is inaccessible unless if its a child process of debugged process
+				RDebugPid *dbg_pid = __build_debug_pid (pe.th32ProcessID, b ? rio->ph : NULL, pe.szExeFile);
```

Comment:
wrong type cast: `__build_debug_pid` expects `const char* name` but if you compile r2 with UNICODE support the string will be truncated